lib/runner/extract-runnable-items.js

```diff
@@ -182,16 +195,63 @@ var sdk = require('postman-collection'),
         }
 
         // extract runnable items from the searched items.
-        for (i = 0, ii = items.length; i < ii; i++) {
-            runnableItems = runnableItems.concat(flattenNode(items[i]));
+        for (i = 0, ii = items.members.length; i < ii; i++) {
+            runnableItems = runnableItems.concat(flattenNode(items.members[i]));
+        }
+
+        callback(null, runnableItems, collection);
+    },
+
+    /**
+     * Finds items or item groups in a collection with matching list of ids or names.
+     *
+     * @note runnable items follows the order of entrypoints
+     *
+     * @param {Collection} collection
+     * @param {Object} options
+     * @param {Array<String>} [options.execute]
+     * @param {Function} callback
+     */
+    lookupByOrder = function (collection, options, callback) {
+        var entrypoints = options.execute,
+            entrypointLookup = {},
+            runnableItems = [],
+            items,
+            i,
+            ii;
+
+        if (!(Array.isArray(entrypoints) && entrypoints.length)) {
+            return callback(null, []);
+        }
+
+        // add temp reference for faster lookup of entry-point name/id.
+        // entry-points with same name/id will be ignored.
+        for (i = 0, ii = entrypoints.length; i < ii; i++) {
+            entrypointLookup[entrypoints[i]] = true;
+        }
+
+        items = findItemsOrGroups(collection, entrypointLookup, true);
+
+        // at this point of time, we should have traversed all items mentioned in entrypoint and created a linear
+        // subset of items. However, if post that, we still have items remaining in lookup object, that implies that
+        // extra items were present in user input and corresponding items for those do not exist in collection. As such
+        // we need to bail out if any of the given entry-point is not found.
+        if (Object.keys(entrypointLookup).length) {
```

Comment:
I see what you mean now @codenirvana. even one extra item causes a bailout. but seems like that it is the acceptable product behaviour for now.